### PR TITLE
CID 242787 CID 242788 CID 242789 fix coverity issues

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

### CODE REVIEW

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c [hunk 1]
```diff
@@ -57,22 +57,27 @@ static ssize_t hwmon_sensor_show(struct device *dev,
...
+	}
+
+	if (field_id == SENSOR_MAX_VAL) {
 		memcpy(value, &sdm->sensor_data[repo_id][buf_index], buf_len);
 		return sprintf(buf, "%u\n", *value);
-	default:
-		return sprintf(buf, "%d\n", 0);
 	}
+
+	return sprintf(buf, "N/A\n");
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:80]: why do you need to change switch statement to if-else statement here?

Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:80]: Hi @maxzhen switch statement expecting "break", but we used return. So, hence the issue.

**Comment by keryell** [comment]
How the build passed on this while it does not compile? :-(
I am sending a PR...

**Comment by dbenusov-xilinx** [comment]
I am also not able to install the with apt-get reinstall.
Here is the error message:
```
Building initial module for 5.4.0-91-generic
ERROR: Cannot create report: [Errno 17] File exists: '/var/crash/xrt.0.crash'
Error!  Build of xocl.ko failed for: 5.4.0-91-generic (x86_64)
Consult the make.log in the build directory
/var/lib/dkms/xrt/2.13.0/build/ for more information.
```
This is the make.log file output (just the important bit)
```
  CC [M]  /var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.o
/var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.c: In function ‘hwmon_sysfs_create’:
/var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.c:94:28: warning: passing argument 1 of ‘strcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   94 |  strcpy(iter->dev_attr.attr.name, sysfs_name);
      |         ~~~~~~~~~~~~~~~~~~~^~~~~
In file included from ./include/linux/bitmap.h:9,
                 from ./include/linux/cpumask.h:12,
                 from ./arch/x86/include/asm/cpumask.h:5,
                 from ./arch/x86/include/asm/msr.h:11,
                 from ./arch/x86/include/asm/processor.h:21,
                 from ./include/linux/mutex.h:19,
                 from ./include/linux/kernfs.h:12,
                 from ./include/linux/sysfs.h:16,
                 from ./include/linux/kobject.h:20,
                 from ./include/linux/cdev.h:5,
                 from ./include/drm/drmP.h:36,
                 from /var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/../xocl_drv.h:35,
                 from /var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.c:18:
./include/linux/string.h:482:37: note: expected ‘char *’ but argument is of type ‘const char *’
  482 | __FORTIFY_INLINE char *strcpy(char *p, const char *q)
      |                               ~~~~~~^
/var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.c: In function ‘hwmon_sdm_probe’:
/var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.c:486:12: error: expected ‘;’ before ‘:’ token
  486 | goto failed:
      |            ^
      |            ;
/var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.c:486:1: error: label ‘failed’ used but not defined
  486 | goto failed:
      | ^~~~
make[3]: *** [scripts/Makefile.build:270: /var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf/../subdev/hwmon_sdm.o] Error 1
make[2]: *** [Makefile:1762: /var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf] Error 2
make[2]: Leaving directory '/usr/src/linux-headers-5.4.0-91-generic'
make[1]: *** [Makefile:133: all] Error 2
make[1]: Leaving directory '/var/lib/dkms/xrt/2.13.0/build/driver/xocl/userpf'
make: *** [Makefile:2: all] Error 2

```

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c [hunk 2]
```diff
@@ -468,27 +473,27 @@ static int hwmon_sdm_probe(struct platform_device *pdev)
...
-	if (err)
+	if (err) {
 		xocl_err(&pdev->dev, "hwmon_sdm hwmon_sysfs is failed, err: %d", err);
+		goto failed;
+	}
 
 	xocl_info(&pdev->dev, "hwmon_sdm driver probe is successful");
 	return 0;
+
+goto failed:
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:486]: I believe this is supposed to be
`failed:` instead of `goto failed:`

**Comment by keryell** [comment]
@dbenusov-xilinx This should be fixed by https://github.com/Xilinx/XRT/pull/6089

**Pull request merged**